src/servers/graphql-server.ts

```diff
@@ -142,31 +142,39 @@ export const sessionContext = ({
   return addAttributesToCurrentSpanAndPropagate(
     {
       "token.sub": tokenPayload?.sub,
+      "token.iss": tokenPayload?.iss,
       [SemanticAttributes.HTTP_CLIENT_IP]: ip,
     },
     async () => {
       // note: value should match (ie: "anon") if not an accountId
       // settings from dev/ory/oathkeeper.yml/authenticator/anonymous/config/subjet
-      const maybeUserId = checkedToUserId(tokenPayload?.sub || "")
+      const sub = tokenPayload?.sub || ""
+
+      const maybeUserId = checkedToUserId(sub)
+      const maybeDeviceId = checkedToDeviceId(sub)
```

Comment:
ok I now understand why we have all those unexpected, and what feel weirdly designed, casting between DeviceId and UserId. 

this line/file is the main reason: we can receive from Oathkeeper either a UserId or a DeviceId from the same field.

I don't believe we are using Oathkeeper properly here. We should not receive both a DeviceId and a UserId from the sub. we should always receive a UserId from kratos. 

the backend should, at this stage, not have to bother what type of identity it's receiving. it should always be the same.

how can we manage this properly?

We can use an [hydrator](https://www.ory.sh/docs/oathkeeper/pipeline/mutator#hydrator) as an additional mutator in the oathkeeper pipeline. we should conditional call a new API endpoint in the backend when the request comes from a JWT, and make the translation deviceId to UserId, so that the request at the backend always comes from a kratos userId. this way there would be no need for custom logic. 

-- 
that said, there might be one reason to not do this. we previously had a lot of 503 when we had a custom authenticator that was pointing to the backend. we never really debug the issue, and instead got rid of the backend request we remove the legacy JWT.